backend/stakepoold/rpcclient.go

```diff
@@ -134,30 +136,19 @@ func walletGetTickets(spd *stakepool.Stakepoold) (map[chainhash.Hash]string, map
 
 	log.Info("Calling GetTickets...")
 	timenow := time.Now()
-	tickets, err := spd.WalletConnection.RPCClient().GetTickets(false)
+	tickets, err := spd.WalletConnection.RPCClient().GetTickets(ctx, false)
 	log.Infof("GetTickets: took %v", time.Since(timenow))
 
 	if err != nil {
 		log.Warnf("GetTickets failed: %v", err)
 		return ignoredLowFeeTickets, liveTickets, err
 	}
 
-	type promise struct {
-		rpcclient.FutureGetTransactionResult
-	}
-	promises := make([]promise, 0, len(tickets))
-
-	log.Debugf("setting up GetTransactionAsync for %v tickets", len(tickets))
-	for _, ticket := range tickets {
-		// lookup ownership of each ticket
-		promises = append(promises, promise{spd.WalletConnection.RPCClient().GetTransactionAsync(ticket)})
-	}
-
```

Comment:
the previous code started all rpc calls without waiting for any of them to complete, and then waited on each of those calls finishing before processing the result.  this updated version will serialize all of the calls which would dramatically slow down this procedure.  in order to regain the async nature of this code, each call should be executed as its own goroutine, saving the results to a slice (probably) where each index of that slice corresponds to a started call.  You can use a errgroup or waitgroup to wait until all of the calls have finished (either in success or error) before iterating through all of the errors and returned results below.